streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -584,6 +579,7 @@ public boolean test(final K1 key, final V1 value) {
         String name = topology.newName(LEFTJOIN_NAME);
 
         topology.addProcessor(name, new KStreamKTableLeftJoin<>((KTableImpl<K, ?, V1>) other, joiner), this.name);
+        topology.connectProcessorAndStateStores(name, other.getStoreName());
```

Comment:
`KTable` may not be materialized and hence its `getStoreName` may be returning null. In its `KStreamKTableLeftJoin` implementation we are using `KTableValueGetter` which may be fetching data from the ancestor's materialized store.

In fact, do we need to connect this processor with the `KTable`'s store? the processor does not directly access the store at all.